python/lsst/faro/measurement/TractTableMeasurementTasks.py

```diff
@@ -218,21 +221,69 @@ def run(
 
         self.log.info("Measuring %s", metricName)
 
+        # Validation to require that gri bands are in the input set
+        if set(currentBands).issuperset(set(['g', 'r', 'i'])) is False:
+            self.log.warn("Data in gri bands required for wPerp calculation.")
+            return Struct(measurement=Measurement(metricName, np.nan * u.mmag))
+
         # If accessing objectTable_tract, we need to append the band name at
         #   the beginning of the column name.
         if currentBands is not None:
             prependString = currentBands
         else:
             prependString = None
 
+        # import pdb; pdb.set_trace()
+
         # filter catalog
         catalog = selectors.applySelectors(catalog,
                                            self.config.selectorActions,
                                            currentBands=currentBands)
 
-        extVals = extinction_corr(catalog, currentBands)
-
-        # Next, figure out how to call Sophie's fit function(s).
-        # The main issue is figuring out how to give the bands in the order we need?
-        fitParams = stellarLocusFit(xs, ys, self.config.stellarLocusFitDict)
+        # Filter based on configured r-mag limits:
+        # rmag_tmp = (catalog.r_psfFlux.values*u.nJy).to(u.ABmag).value
+        # magfilter = (rmag_tmp < self.config.faint_rmag_cut) &\
+        #             (rmag_tmp > self.config.bright_rmag_cut)
+        # catalog = catalog[magfilter]
+
+        # Create a SkyCoord object to get the extinction:
+        sc = SkyCoord(catalog['coord_ra']*u.deg, catalog['coord_dec']*u.deg)
+
+        extVals = extinctionCorrTable(sc, currentBands)
+
+        g0 = (catalog.g_psfFlux.values*u.nJy).to(u.ABmag).value - extVals['A_g']
+        r0 = (catalog.r_psfFlux.values*u.nJy).to(u.ABmag).value - extVals['A_r']
+        i0 = (catalog.i_psfFlux.values*u.nJy).to(u.ABmag).value - extVals['A_i']
+
+        # Call the stellar locus fit function:
+        p1vals, p2vals, fitParams = stellarLocusFit(g0-r0, r0-i0,
+                                                    self.config.stellarLocusFitDict)
+
+        # Ivezic+2004 defines wPerp between -0.2 < p1 < 0.6.
+        # Also, do some expanded color cuts to get rid of large color outliers that
+        #   skew the rms.
+        magcushion = 0.2
+        okp1 = (p1vals > -0.2) & (p1vals < 0.6) &\
+               ((r0-i0) < fitParams['yMax']+magcushion) &\
+               ((r0-i0) > fitParams['yMin']-magcushion) &\
+               ((g0-r0) < fitParams['xMax']+magcushion) &\
+               ((g0-r0) > fitParams['xMin']-magcushion)
+
```

Comment:
We shouldn't hard code this stuff into the function, we're going to need different values for different cameras and hard coding it makes it more difficult to keep analysis_drp and faro in sync. Some of this needs to be refactored into a single place. I think we discussed a new repo called analysis that both faro and analysis_drp (and future analysis_ap) can import from.

Reply:
I don't understand why this is necessary.

Reply:
> We shouldn't hard code this stuff into the function, we're going to need different values for different cameras.

I assume you're referring to the p1 limits. As far as I know, they are defined once in Table 3 of Ivezic+2004. If we are going to adopt a way of varying these for different cameras, then we will need to decide on how they will be defined. That should be a conversation for the future, and if we do define limits that vary per camera, we should write down explicitly how they are defined and calculated.

Reply:
> I don't understand why this is necessary. Why use cuts from a different survey and then add padding anyway? 

The color cuts were needed because there are huge color outliers that still fit within the p1 ranges (e.g., you can have p1=0.0, and p2=2.0 for an object at large r-i color; we don't want those extreme stars in the sample). I want to remove them from the final stats. I apply padding to the color ranges because I don't want them to supersede the ranges in p1. [We could instead use sigma-clipping of the p2 values, but when I tested it the large color outliers were skewing the median p2 values, so that the sigma clipping wasn't effectively the sample to the region of interest.

> Also you follow Ivezic 2004 here but then don't for the parallel component.

Not sure I understand your point -- I follow the p1 ranges from Ivezic+2004, Table 3 (i.e., -0.2 < p1 < 0.6).